### PR TITLE
added tests for issues and some generic algebra tests

### PR DESCRIPTION
**Pull request by karlwessel**
I added two files with more tests.

`issuestests.jl` contains tests cases that cover any issues reported on github (including currently open ones). These should assure that they are not reintroduced in a later commit. A good practice probably would be to add a tests case for any issue to that file before fixing that issue. 

`generictests.jl` contains a collection of tests that check mathematical properties which should hold for any geometric algebra. The collection can be executed on different signatures over different fields.

Although those test improve the coverage somewhat they are not especially written to do that. For that I would rather write another file with tests which only make sure that as much lines as possible are covered, without caring to much about the output or the sensibility of the test. These would at least assure that there are no syntax errors.

### CODE REVIEW

**Comment by utensil** [comment]
@karlwessel Just commented on the code a little bit. Your tests look well organized and you are very observant to raise the issues. :+1: 

I was also thinking about porting the tests in https://github.com/pygae/GAlgebra.jl/blob/master/test/runtests.jl to Grassmann, including sanity checks and many Geometric Algebra identities. More tests are planned, like Geometric Calculus and check the properties of other algebra structures (groups, ideals etc. ) that GA can generate.

P.S. [GAlgebra.jl](https://github.com/pygae/GAlgebra.jl) is a symbolic GA library in Julia (aimed to be more than just a wrapper for [GAlgebra](https://github.com/pygae/galgebra)), it would be nice that it also comes in handy for you. :smile:

File: test/issuestests.jl [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    
+    @test v∅*v∞ == -1 - v∞∅
+    @test_broken v∅*(-v∞) == 1 + v∞∅
+    
+    a = v∅*basis(-v∞)
+    @test a == -1 - v∞∅
+    @test_broken SValue{V}(-1, a)
+end
+
+@testset "Issue 22: Error in MultiVector constructor for Blades" begin
```
Comment (chakravala) [inline test/issuestests.jl]: this has been fixed now, and is no longer a broken test

**Review by chakravala — CHANGES_REQUESTED** [review]
Thanks for your contribution, please move the `isapprox` and `scalar` methods to `src/multivectors.jl` into the section labeled `## Generic` around lines 300 and `export scalar`

The problem for #22 has been resolved in https://github.com/chakravala/Grassmann.jl/commit/0c6889221d6762d101863b1eb09ad23044f8ac55 so it is no longer a broken test

File: test/generictests.jl [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+for 𝔽 in [Float64]
+    e = one(𝔽)
+    α = rand(𝔽)
+    @test α*e == α
+    @testset "Field: $(𝔽)" begin 
+        for G in [V"+++", S"∞+", S"∅+", V"-+++",
+                  S"∞∅+" # is currently broken for associativity
+                  ]
+            @testset "Algebra: $(G)" begin
+                dims = length(G)
```
Comment (chakravala) [inline test/generictests.jl]: you can use `ndims(G)` for that also

File: test/issuestests.jl (hunk 1 above)
Comment (karlwessel) [reply test/issuestests.jl]: I still get an error for creating an SBlade from a Basis. See #22

File: test/generictests.jl (hunk 2 above)
Comment (karlwessel) [reply test/generictests.jl]: Ah, thanks for the hint. There are probably some more points where I could use some more direct methods like that.

**Review by chakravala — APPROVED** [review]

**Comment by chakravala** [comment]
@karlwessel are there any additional tests planned, or not? p.s. @utensil

**Comment by karlwessel** [comment]
> @karlwessel are there any additional tests planned, or not? p.s. @utensil

Not at the moment. I write the tests whenever something in my own code doesn't work as expected. Implementing tests for the properties of GA helps me by telling me that I didn't understand some property correctly yet or by telling me that it isn't implemented correctly yet. 

There are also the pure coverage related tests that one should implement. Like my tests for the `isapprox` and `scalar` method, which one probably should move to a `multivectorstest.jl`. But that is a rather boring work :-). And since you said that the API isn't stable yet it probably makes more sense the write those for the final API.

So there might be more tests coming but not for this PR I would say.

**Pull request merged**